File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.py
class Solution:
    def findWinners(self, matches: List[List[int]]) -> List[List[int]]:
        won = set()
        loss = set()
        more = set()
        for arr in matches :
            if arr[0] not in loss and arr[0] not in more :
                won.add(arr[0])
            if arr[1] in loss or arr[1] in more :
                loss.discard(arr[1])
                more.add(arr[1])
            else :
                loss.add(arr[1])
                if arr[1] in won : won.discard(arr[1])
        res = []
        res.append(list(won))
        res.append(list(loss))
        res[0].sort()
        res[1].sort()
        return res

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        Set<Integer> won = new HashSet<>();
        Set<Integer> loss = new HashSet<>();
        Set<Integer> more = new HashSet<>();
        List<List<Integer>> res = new ArrayList<>();
        for(int[] arr : matches) {
            if(loss.contains(arr[1]) || more.contains(arr[1])) {
                loss.remove(arr[1]);
                more.add(arr[1]);
            }
            else{
                loss.add(arr[1]);
                if(won.contains(arr[1])) won.remove(arr[1]);
            }
            if(!loss.contains(arr[0]) && !more.contains(arr[0])) {
                won.add(arr[0]);
            }
        }
        res.add(new ArrayList<>(won));
        res.add(new ArrayList<>(loss));
        Collections.sort(res.get(0));
        Collections.sort(res.get(1));
        return res;
    }
}